src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2362 @@
+            common_channel, mix_channel. In flatten, patch embedding encodes the patch information across all channels.
+            In common_channel mode, patch embedding is independent of channels (Channel Independece). In mix_channel,
+            we follow channel independence, but in addition to patch and feature mixing, we also do channel mixing.
+            Defaults to "common_channel".
+        gated_attn (bool, *optional*): Enable Gated Attention. Defaults to False.
+        ffn (str, *optional*): MLP mode. Allowed values: mlp, gmlp. gmlp is not preferred. Defaults to "mlp".
+        self_attn (bool, *optional*): Enable Tiny self attention in addition to MLP mixing. Defaults to False.
+        self_attn_heads (bool, *optional*): Self attention heads. Defaults to 1.
+        norm_mlp (str, *optional*): Norm layer (BatchNorm or LayerNorm). Defaults to LayerNorm.
+
+    """
+
+    def __init__(
+        self,
+        num_patches: int,
+        num_features: int = 16,
+        in_channels: int = 3,
+        expansion_factor: int = 2,
+        dropout: float = 0.2,
+        mode: str = "common_channel",
+        gated_attn: bool = False,
+        ffn: str = "mlp",
+        self_attn: bool = False,
+        self_attn_heads: int = 1,
+        norm_mlp: str = "LayerNorm",
+    ):
+        super().__init__()
+        self.patch_mixer = PatchMixerBlock(
+            num_patches=num_patches,
+            num_features=num_features,
+            expansion_factor=expansion_factor,
+            dropout=dropout,
+            mode=mode,
+            gated_attn=gated_attn,
+            ffn=ffn,
+            self_attn=self_attn,
+            self_attn_heads=self_attn_heads,
+            norm_mlp=norm_mlp,
+        )
+        self.feature_mixer = FeatureMixerBlock(
+            num_features=num_features,
+            expansion_factor=expansion_factor,
+            dropout=dropout,
+            gated_attn=gated_attn,
+            ffn=ffn,
+            mode=mode,
+            norm_mlp=norm_mlp,
+        )
+        # define a cross series mixer
+
+        self.mode = mode
+        if mode == "mix_channel":
+            self.channel_feature_mixer = ChannelFeatureMixerBlock(
+                num_features=num_features,
+                in_channels=in_channels,
+                expansion_factor=expansion_factor,
+                dropout=dropout,
+                mode=mode,
+                gated_attn=gated_attn,
+                ffn=ffn,
+            )
+
+    def forward(self, x):
+        # x.shape == (batch_size, num_patches, num_features)
+        if self.mode == "mix_channel":
+            # x = self.channel_token_mixer(x)
+            x = self.channel_feature_mixer(x)
+
+        x = self.patch_mixer(x)
+        x = self.feature_mixer(x)
+
+        # x.shape == (batch_size, num_patches, num_features)
+        return x
+
+
+class PatchTSMixerBackbone(nn.Module):
```

Comment:
```suggestion
class PatchTSMixerEncoder(nn.Module):
```
We typically use the term `Encoder` for this

Reply:
Hi @NielsRogge , thanks for the suggestion. We already have a `PatchTSMixerEncoder` which is used by the `PatchTSMixerModel` class. The `PatchTSMixerModel` is exposed. On the other hand, `PatchTSMixerBackbone` is used by the `PatchTSMixer` class which is not exposed (just internally used).

Reply:
I think, the work backbone is confusing here. I renamed it to PatchTSMixerBlock